### PR TITLE
Prototype improve search interface #7423

### PR DESCRIPTION
**Pull request by LachyP**
The original issue involved improving the search interface by using OmniSearch. Issue #7423
With the idea that eventually it would look something like this:
image

The following has been added by me and my team (@shameenkals @JasonBourne999 @mchellelm @artherdunger10):
- When a user selects an author from the autocomplete dropdown, "author=xyz" is added to the search text
- The heading "Author/Editor" appears above the search term results
- A function in SearchPopupSkin that can combine multiple observable lists together
- The code has been written for an autocompleter that searches by title instead of author, however, it is commented out at the moment

Early prototype looks like this:
image

Current Problems:
- Had to make SearchPopupSkin a non-static class since it wouldn't work otherwise
- Heading functionality relies on the fromString() method, however, all terms should ideally be converted to strings before being added to the ListView
- Haven't yet figured out how to search multiple different categories simultaneously.
- Haven't yet figured out how to alter the appearance of the autocomplete dropdown with css.

The development of this feature isn't done, however, we figure this will be a good space for my team to discuss with other developers, seek advice, and see what direction this PR is heading towards.

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by mchellelm** [comment]
Hi, I'm having some issues figure out what files and classes to change to make changes on the stylesheet. Any tips would be appreciated!

**Comment by tobiasdiez** [comment]
That looks already pretty good!

For the styling, did you try to style the class auto-complete-popup in the main css file? Somewhere hidden in the javafx docs there is also an overview of the styleable subcomponents. For this you could also try https://github.com/JonathanGiles/scenic-view, but I'm not sure if it works with the recent jdk version that we use.

File: src/main/java/org/jabref/gui/autocompleter/StringStringConverter.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    @Override
+    public String toString(String string) {
+        return string;
+    }
+
+    @Override
+    public String fromString(String string) {
+        return string;
+    }
```
Comment (calixtus) [inline src/main/java/org/jabref/gui/autocompleter/StringStringConverter.java:17]: This seems to be a bit silly, doesn't it? Maybe there is a generic-converter for it in javax.util.stringconverter?

**Comment by calixtus** [comment]
Any update here? Otherwise I would close it for now, until someone eventually wants to continue this one...

**Pull request closed without merging**